play/src/containers/GameInProgressContainer.tsx

```diff
@@ -48,21 +57,34 @@ export default function GameInProgressContainer(
             retry={retry}
             handleRetry={handleRetry}
           />
-          <HowToPlay mode={HowToPlayMode.ERROR} />
+          <Lobby mode={LobbyMode.ERROR} />
         </>
       );
     }
     // if loading, display loading message on bottom of How to Play page
-    if (subscription.isLoading) return <HowToPlay mode={HowToPlayMode.LOADING} />;
+    if (subscription.isLoading) return <Lobby mode={LobbyMode.LOADING} />;
     // if waiting for teacher, display waiting message on How to Play page
-    return <HowToPlay mode={HowToPlayMode.READY} />;
+    return <Lobby mode={LobbyMode.READY} />;
   }
   // if teacher has started game, pass updated gameSession object down to GameSessionSwitch
   return (
     <GameSessionSwitch
+      isRejoin={subscription.isRejoin}
       pregameModel={pregameModel}
       gameSession={subscription.gameSession}
       {...props}
     />
   );
 }
+
+export function GameInProgressContainerLoader() {
```

Comment:
I've renamed this as `LocalModelLoader` because with the rejoin functionality added, there's less distinction between pregame and gameinprogress with respect to localstorage. I've also renamed `pregameModel` and `PregameModel` to `LocalModel` and have annotated that return type here. 

This loader function is used here (`App.tsx`):
`<Route
        path="/game"
        element={<GameInProgressContainer apiClient={apiClient} />}
        loader={LocalModelLoader}
      />`

to preload the data at either initial game load or whenever a player rejoins. I liked keeping the loader at the bottom of the `GameInProgressContainer` because it's in this component (and only this component) that it's used. Similar to the (simplified) example from the [docs](https://reactrouter.com/en/main/hooks/use-loader-data). Let me know if you think this is confusing though.